my-app/src/components/AllButtons.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './AllButtons.css';
+
+// generic component for a button 
+export default function GenericButton ({
+    label, 
+    onClick, 
+    style = {},
+    className = "generic-button"
+}) { return (
+    <button 
+        className={className} 
+        style={style} 
```

Comment:
A great use case for generic components is standardizing the styles for your app. 
You can make your GenericButton have a particular style, so every time you want a button in your app, you can use `<GenericButton/>` instead of `<button />` and it will automatically have all of your preferred button styling applied!
To do this, your `GenericButton` wouldn't accept a `style` or `className` prop, only `label` and `onClick`. You hardcode the style and/or classname inside the `GenericButton` component to enforce a specific style across all buttons in your app.
Then when you use `GenericButton`, you don't have to pass the styling and class name to get the styling you want.

Reply:
Sometimes people want multiple different kinds of buttons, and each button has its own slightly different style. In which case they would just create multiple variants of the generic button component.